### PR TITLE
Fix relations for os, added relations tests

### PR DESCRIPTION
**Pull request by Marginy605**
Merge after https://github.com/datalens-tech/datalens-ui/pull/351 
- Fix flag check util
- Add relations tests, fix test data in dash

### CODE REVIEW

File: src/shared/constants/qa/components.ts [hunk 1]
```diff
@@ -150,3 +150,7 @@ export const enum CreateEntityButton {
 export enum DashTabsQA {
     Root = 'dash-tabs',
 }
+
+export enum Select {
```
Comment (mournfulCoroner) [inline src/shared/constants/qa/components.ts:154]: `SelectQA`? It will be easier to find and parse

File: tests/suites/dash/base/loading.test.ts [hunk 2]
```diff
@@ -113,7 +119,7 @@ datalensTest.describe('Dashboards - Widget Downloads', () => {
         },
     );
     datalensTest(
-        "Loading charts that didn't get into the viewport when opening links",
+        "Load ing charts that didn't get into the viewport when opening links",
```
Comment (goshander) [inline tests/suites/dash/base/loading.test.ts:122]: Extra space?

Comment (mournfulCoroner) [inline tests/suites/dash/base/loading.test.ts:122]: Extra space

**Review by goshander — APPROVED** [review]

File: tests/page-objects/dashboard/DashboardPage.ts [hunk 3]
```diff
@@ -177,6 +184,36 @@ class DashboardPage extends BasePage {
...
+        // waiting for the dialog to open, specify the name, save
+        await this.workbookDialogFillAndSave(this.page, newDashName);
+
+        await this.page
+            .locator(`${slct(WorkbookPage.ListItem)}:has-text('${newDashName}')`)
+            .click();
+    }
+
+    // Fill in the input with the name of the entity being created in the EntryDialog (the dialog that appears when saving entities) and click the "Create" button
+    async workbookDialogFillAndSave(page: Page, entryName: string) {
```
Comment (mournfulCoroner) [inline tests/page-objects/dashboard/DashboardPage.ts:205]: There is a similar function here https://github.com/datalens-tech/datalens-ui/blob/main/tests/page-objects/workbook/DialogCreateEntry.ts#L28

File: tests/page-objects/dashboard/DashboardPage.ts [hunk 4]
```diff
@@ -432,6 +481,58 @@ class DashboardPage extends BasePage {
...
+        await selectorElem.click();
+    }
+
+    async setupNewLinks({
+        linkType,
+        widgetElem,
+        firstParamName,
+        secondParamName,
+    }: {
+        linkType:
```
Comment (mournfulCoroner) [inline tests/page-objects/dashboard/DashboardPage.ts:505]: It seems that here you can use just `DashRelationTypes` as type

**Review by mournfulCoroner — APPROVED** [review]

**Comment by Marginy605** [comment]
Locally all tests passed:
image

But tests on github actions start with changes from docker image (old one). So we need to up datalens-ui version in main datalens repository with code changes (flags, qa, etc). And only after that test will pass.

**Pull request merged**